solver/src/settlement_submission/flashbots_api.rs

```diff
@@ -30,7 +30,26 @@ impl FlashbotsApi {
         let status = response.status();
         let body = response.text().await?;
         ensure!(status.is_success(), "status {}: {:?}", status, body);
-        Ok(body)
+
+        match serde_json::from_str::<jsonrpc_core::Output>(&body) {
+            Ok(body) => {
+                let bundle_id = if let jsonrpc_core::Output::Success(ref x) = body {
+                    x.result.as_str().unwrap_or_default()
+                } else {
+                    Default::default()
+                };
+                tracing::debug!(
+                    "flashbots bundle id: {}",
+                    serde_json::to_string(&bundle_id)
+                        .unwrap_or_else(|err| format!("error: {:?}", err)),
+                );
+                Ok(bundle_id.to_string())
+            }
+            Err(err) => {
+                tracing::debug!("failed to submit: {}", err);
+                Err(anyhow!("failed to submit"))
```

Comment:
Maybe also propagate the error in the message we bubble up to the caller (otherwise we need to enable debug logs to find the error message)

Reply:
Yes, fixed.